sensor/kubernetes/listener/resources/convert.go

```diff
@@ -461,7 +461,7 @@ func (w *deploymentWrap) updatePortExposureFromServices(svcs ...serviceWithRoute
 }
 
 func (w *deploymentWrap) updatePortExposure(svc serviceWithRoutes) {
-	if svc.selector.Matches(labels.Set(w.PodLabels)) {
+	if svc.selectorWrapper.Matches(labelWrapper{labels.Set(w.PodLabels), uint(len(w.PodLabels))}) {
```

Comment:
Look, now when we have the `labelWrapper` we can use a constructor and make this less ugly by skipping the `labels.Set`:

```go
func WrapLabels(m map[string]string) labelWrapper {
...
}
```

and then use it like this:

```go
svc.selectorWrapper.Matches(WrapLabels(w.PodLabels))
```